packages/core/test/polygon/e2e.js

```diff
@@ -0,0 +1,176 @@
+const { ZERO_ADDRESS, interfaceName, RegistryRolesEnum, didContractThrow } = require("@uma/common");
+const { assert } = require("chai");
+const TruffleAssert = require("truffle-assertions");
+
+const { utf8ToHex, toWei, hexToUtf8 } = web3.utils;
+
+const StateSync = artifacts.require("StateSyncMock");
+const FxChild = artifacts.require("FxChildMock");
+const FxRoot = artifacts.require("FxRootMock");
+const OracleChildTunnel = artifacts.require("OracleChildTunnel");
+const OracleRootTunnel = artifacts.require("OracleRootTunnelMock");
+const Finder = artifacts.require("Finder");
+const Registry = artifacts.require("Registry");
+const MockOracle = artifacts.require("MockOracleAncillary");
+const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
+
+contract("Polygon <> Ethereum Tunnel: End-to-End Test", async (accounts) => {
+  const owner = accounts[0];
+  const systemSuperUser = accounts[1];
+  const checkpointManager = accounts[2];
+  const rando = accounts[3];
+
+  let stateSync;
+  let fxChild;
+  let fxRoot;
+  let oracleChild;
+  let oracleRoot;
+
+  // Oracle system:
+  let finder;
+  let identifierWhitelist;
+  let registry;
+  let mockOracle;
+
+  const testIdentifier = utf8ToHex("TEST");
+  const testTimestamp = 100;
+  const testAncillaryData = utf8ToHex("key:value");
+  const testPrice = toWei("0.5");
+  let expectedStampedAncillaryData; // Can determine this after contracts are deployed.
+  const expectedStateId = "1";
+
+  before(async function () {
+    finder = await Finder.deployed();
+    identifierWhitelist = await IdentifierWhitelist.deployed();
+    registry = await Registry.deployed();
+    mockOracle = await MockOracle.new(finder.address, ZERO_ADDRESS);
+
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.Registry), registry.address);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.Oracle), mockOracle.address);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.IdentifierWhitelist), identifierWhitelist.address);
```

Comment:
if you are importing these contracts from the truffle artifacts using the `.deployed` syntax why do you need to `changeImplementationAddress` on them? are they not registered from the migrations like in other tests in core?

Reply:
Good catch!